src/cpp/src/visual_language/embedding_model.cpp

```diff
@@ -41,13 +58,18 @@ EmbeddingsModel::EmbeddingsModel(const std::string& model,
     merge_postprocess(m_model, scale_emb);
 
     ov::CompiledModel compiled_model = core.compile_model(m_model, device, properties);
-    m_request = compiled_model.create_infer_request();
+    std::tie(m_request, m_cpu_tensor, m_remote_tensor) = init(compiled_model);
 }
 
-ov::Tensor EmbeddingsModel::infer(ov::Tensor input_idx) {
+ov::Tensor EmbeddingsModel::infer(const ov::Tensor& input_idx, bool return_remote_tensor) {
     OPENVINO_ASSERT(m_request, "Text embeddings decoder model must be compiled first. Cannot infer non-compiled model");
 
     m_request.set_input_tensor(input_idx);
+    if (return_remote_tensor) {
+        m_request.set_output_tensor(m_remote_tensor);
```

Comment:
@Wovchena is it required to set tensor on each infer? Can it be set just once during `init`?

Reply:
discussed